### ISSUE TITLE
Functionality-breaking errors on Gitpod - Cannot read properties of null (reading 'content')

### ISSUE THREAD

**Issue by Xanthus** [ionide/ionide-vscode-fsharp#1811]
_Edit: Initially this issue was about getting stuck at 'Loading projects' when opening the Gitpod workspace. This wasn't the problem I initially had but I couldn't reproduce that at the time of creating the issue. I've since figured out that Gitpod caches VS Code extension settings to a certain extent and that caused the 'Loading projects' issue. Once the settings were cleared, I got back to the original issue described below of uncaught errors that broke a lot of functionality in Gitpod._

### Describe the bug

Ionide doesn't fully work when opening the repo on Gitpod as a new workspace. It activates and the solution explorer works but there are no tooltips, no syntax checking, inlay hints, jump to definition, etc. There is an uncaught error on every attempt to get a tooltip: `Cannot read properties of null (reading 'content')`.

Also, [this line](https://github.com/ionide/ionide-vscode-fsharp/blob/428764e236422128028590102cb1aa2903b7db12/build/Program.fs#L351) of code also appears not to be invoked/work correctly as launching the debugger throws an error of `Cannot activate the 'Ionide for F#' extension because it depends on an unknown 'ms-dotnettools.csharp' extension.` After manually updating `release/package.json` to reference the updated C# extension name with `"extensionDependencies": ["muhammad-sammy.csharp"],`  the extension activates in the extension development host but hovering to get a tooltip on code throws the same error as in the main VS Code window: 
```
rejected promise not handled within 1 second: TypeError: Cannot read properties of null (reading 'content')
stack trace: TypeError: Cannot read properties of null (reading 'content')
	at /workspace/ionide-vscode-fsharp/release/fsharp.js:48567:224
```

Digging into the referenced JS line, it seems the issue is [here](https://github.com/ionide/ionide-vscode-fsharp/blob/22c26a5295c5368c2f6ba9793a8d1f45f1f381a6/src/Core/LanguageService.fs#L255) where the response for a signature request is returned as null:

````fsharp
let signature (uri: Uri) line col =
        match client with
        | None -> Promise.empty
        | Some cl ->
            let req: Types.TextDocumentPositionParams =
                { TextDocument = { Uri = uri.ToDocumentUri }
                  Position = { Line = line; Character = col } }

            cl.sendRequest ("fsharp/signature", req)
            |> Promise.map (fun (res: Types.PlainNotification) -> res.content |> ofJson<Result<string>>)
````

I don't know enough to dig into why the language server is returning null unfortunately.

### Steps to reproduce

_Note that the Gitpod Dockerfile is out of sync with the required SDK version and needs to be [manually updated to 6.0.401](https://github.com/Xanthus/ionide-vscode-fsharp/commit/d0600ffe1e28f99964e5fcfbd38f7d1be58f5027) in order to reproduce this. I haven't created a PR for that as I was hoping to contribute something more meaningful but I can if you'd like._

1) Fork the repo
2) Update the [Gitpod Dockerfile's SDK version](https://github.com/Xanthus/ionide-vscode-fsharp/commit/d0600ffe1e28f99964e5fcfbd38f7d1be58f5027)
3) Create a new Gitpod workspace with the updated repo
4) Once VS Code loads, wait for the build to complete
5) Run the 'Watch' task and wait for the build to complete
6) Update the compiled `package.json` with `"extensionDependencies": ["muhammad-sammy.csharp"],`
7) Run the 'Launch only' debug configuration

You should now have two windows where Ionide won't work properly in either - the first is with the published marketplace extension installed in the main development window and the second is the master branch launched in debug mode.

### Expected behaviour

Ionide should load in both windows - main workspace and debugger.

### Screenshots

Uncaught errors (looks the same in development and debugging windows):

![image](https://user-images.githubusercontent.com/490637/209953882-8ce7ae82-9a42-4fd8-b25a-4da6cf4b28d9.png)

Error in the extension development host on initially launching the debugger:

![image](https://user-images.githubusercontent.com/490637/209952282-591316e9-4b2c-42f8-83b6-c12e7a1aac3e.png)

## Machine info

Gitpod with Ionide 7.4.0 in the editor and master branch in the debugger.

**Comment by cartermp** [comment]
I can't reproduce the above, but I can reproduce a different issue 
<img width="451" alt="image" src="https://user-images.githubusercontent.com/6309070/210010629-2c850b50-af08-4068-9351-74b2efd729b0.png">

Context: https://github.com/cartermp/otel-lang-samples

**Comment by baronfel** [comment]
A few things made @cartermp's sample project work fine for me:

* use adaptive LSP server
* change the TFM of the project to net7.0 (unsure if this is _required_ or not)

![image](https://user-images.githubusercontent.com/573979/210021000-ea9ba2ab-a793-4064-bb4a-503e54ce7529.png)

First thing I'd check in all cases is if the project cracking worked, which will show in a few ways - the `PackageReferences` node will usually have stuff in it, and the `Show project status` command will have a lot of data in it. If either of those aren't true, then binlogs would be next.

**Comment by cartermp** [comment]
Yep, project cracking doesn't work (even if I move to net7 and use the adaptive LSP)

**Comment by baronfel** [comment]
@cartermp when you launch the gitpod, can you cd to the project and run dotnet restore first, before opening the program.fs file?
